zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt

```diff
@@ -60,11 +60,14 @@ class ZiplinePlugin : KotlinCompilerPluginSupportPlugin {
 
     kotlinExtension.targets.withType(KotlinJsIrTarget::class.java).all { kotlinTarget ->
       kotlinTarget.binaries.withType(JsIrBinary::class.java).all { kotlinBinary ->
-        registerCompileZiplineTask(
+        val ziplineCompileTask = registerCompileZiplineTask(
           project = target,
           jsProductionTask = kotlinBinary.asJsProductionTask(),
           extension = ziplineExtension,
         )
+        ziplineCompileTask.configure {
+          it.dependsOn(kotlinBinary.linkTask)
```

Comment:
The error I’m attempting to fix:

```
InvalidUserCodeException: Querying the mapped value of flatmap(provider(task 'compileDevelopmentExecutableKotlinJs', class org.jetbrains.kotlin.gradle.targets.js.ir.KotlinJsIrLink)) before task ':activity:treehouse:compileDevelopmentExecutableKotlinJs' has completed is not supported
```

Reply:
The use of `dependsOn` means either there's a Gradle bug with the use of `Property` or a Kotlin feature request with a lack of `Property` being exposed.

Reply:
Agreed.